test/gurka/test_elevation.cc

```diff
@@ -0,0 +1,233 @@
+  detail::build_pbf(layout, ways, {}, {}, pbf_filename);
+
+  valhalla::gurka::map map;
+  map.nodes = layout;
+  map.config = test::make_config(workdir, {});
+  boost::property_tree::ptree& pt = map.config;
+  pt.put("mjolnir.tile_dir", workdir + "/tiles");
+  pt.put("additional_data.elevation", workdir);
+
+  std::vector<std::string> input_files = {pbf_filename};
+  build_tile_set(pt, input_files, mjolnir::BuildStage::kInitialize, mjolnir::BuildStage::kValidate,
+                 false);
+
+  // try a bunch of routes
+  for (const auto& waypoints : std::vector<std::vector<std::string>>{
+           {"S", "F"},
+           {"C", "N"},
+       }) {
+
+    // get a route with elevation included
+    std::string route_json;
+    auto route = gurka::do_action(valhalla::Options::route, map, waypoints, "bicycle",
+                                  {{"/elevation_interval", "30"}}, {}, &route_json);
+    rapidjson::Document result;
+    result.Parse(route_json.c_str());
+
+    // for each leg
+    for (size_t leg_index = 0; leg_index < waypoints.size() - 1; ++leg_index) {
+      // pull out the shape from the leg
+      auto s =
+          rapidjson::get_child_optional(result, ("/trip/legs/" + std::to_string(leg_index) + "/shape")
+                                                    .c_str());
+      EXPECT_TRUE(s && s->IsString());
+      auto shape = json_escape(s->GetString());
+
+      std::string height_json;
+      std::string request =
+          R"({"height_precision":1,"resample_distance":30,"encoded_polyline":")" + shape + R"("})";
+      auto height = gurka::do_action(valhalla::Options::height, map, request, {}, &height_json);
+
+      // pull out the elevation from the route result leg
+      auto elevation =
+          rapidjson::get_child_optional(result,
+                                        ("/trip/legs/" + std::to_string(leg_index) + "/elevation")
+                                            .c_str());
+      EXPECT_TRUE(elevation && elevation->IsArray());
+      std::vector<float> elevation_along_edges;
+      for (const auto& e : elevation->GetArray()) {
+        elevation_along_edges.push_back(std::round(e.GetFloat() * 10) / 10);
+        std::cout << std::round(e.GetFloat() * 10) / 10 << std::endl;
+      }
+
+      result.Parse(height_json.c_str());
+      // pull out the elevation from the height result
+      elevation = rapidjson::get_child_optional(result, "/height");
+      EXPECT_TRUE(elevation && elevation->IsArray());
+      std::vector<float> elevation_from_skadi;
+      for (const auto& e : elevation->GetArray()) {
+        elevation_from_skadi.push_back(std::round(e.GetFloat() * 10) / 10);
+        std::cout << std::round(e.GetFloat() * 10) / 10 << std::endl;
+      }
+
+      EXPECT_EQ(elevation_along_edges.size(), elevation_from_skadi.size());
+      for (size_t i = 0; i < elevation_along_edges.size(); ++i) {
+        EXPECT_NEAR(elevation_along_edges[i], elevation_from_skadi[i], 0.5f);
```

Comment:
@dnesbitt61 is `0.5f` tolerance ok here?

Reply:
This is fine for now. I suspect we could find cases (esp. long edges) where the accuracy could deviate more, requiring a higher tolerance. Statistically, the delta encoding should do pretty well except the cases where slope exceeds 45 percent.